scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,239 @@
+
+    # GCE variables
+    os.environ['INSTANCE_PREFIX'] = e2e_name
+    os.environ['KUBE_GCE_NETWORK'] = e2e_name
+    os.environ['KUBE_GCE_INSTANCE_PREFIX'] = e2e_name
+
+    # GKE variables
+    os.environ['CLUSTER_NAME'] = e2e_name
+    os.environ['KUBE_GKE_NETWORK'] = e2e_name
+
+    # Get golang into our PATH so we can run e2e.go
+    os.environ['PATH'] += ':/usr/local/go/bin'
+
+    for key, value in job_args.items():
+        os.environ[key] = value
+
+    # dockerized-e2e-runner goodies setup
+    workspace = os.environ.get('WORKSPACE') or os.getcwd()
+    repo = os.environ.get('REPO_DIR') or os.getcwd()
+    artifacts = '%s/_artifacts' % workspace
+    os.makedirs(artifacts)
+
+    # TODO(ixdy): remove when all jobs are setting these vars using Jenkins credentials
+    gce_key = '/var/lib/jenkins/gce_keys/google_compute_engine'
+    if os.path.isfile(gce_key):
+        os.environ['JENKINS_GCE_SSH_PRIVATE_KEY_FILE'] = gce_key
+
+    gce_key_pub = '/var/lib/jenkins/gce_keys/google_compute_engine.pub'
+    if os.path.isfile(gce_key_pub):
+        os.environ['JENKINS_GCE_SSH_PUBLIC_KEY_FILE'] = gce_key_pub
+
+
+    e2e_image_tag = 'v20170104-9031f1d'
+    e2e_image_tag_override = '%s/hack/jenkins/.kubekins_e2e_image_tag' % workspace
+    if os.path.isfile(e2e_image_tag_override):
+        with open(e2e_image_tag_override, 'r') as tag:
+            e2e_image_tag = tag.read()
+
+    docker_env_ignore = [
+      'GOOGLE_APPLICATION_CREDENTIALS',
+      'GOROOT',
+      'HOME',
+      'PATH',
+      'PWD',
+      'WORKSPACE'
+    ]
+
+    docker_extra_args = []
+    if (os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER') and
+            os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '/var/run/docker.sock:/var/run/docker.sock',
+          '-v', '%s:/go/src/k8s.io/kubernetes' % repo,
+          '-e', 'REPO_DIR=%s' % repo,
+          '-e', 'HOST_ARTIFACTS_DIR=%s' % artifacts
+        ])
+
+    if (os.environ.get('JENKINS_USE_LOCAL_BINARIES') and
+            os.environ.get('JENKINS_USE_LOCAL_BINARIES').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '%s/_output":/workspace/_output:ro' % workspace
+        ])
+
+    if 'KUBE_E2E_RUNNER' in os.environ:
+        docker_extra_args.extend([
+          '--entrypoint=%s', os.environ.get('KUBE_E2E_RUNNER')
+        ])
+
+
+    docker_timeout = os.environ.get('DOCKER_TIMEOUT') or '60m'
+
+    # exec
+    container = '%s-%s' % (os.environ.get('JOB_NAME'), os.environ.get('BUILD_NUMBER'))
+
+    print 'Starting...'
+
+    cmd = ['timeout', '-s', 'KILL', docker_timeout, 'docker', 'run', '--rm',
+    # For local test, timeout not work on mac
+    # cmd = ['docker', 'run', '--rm',
+      '--name=%s' % container,
+      '-v', '%s/_artifacts":/workspace/_artifacts' % workspace,
+      '-v', '/etc/localtime:/etc/localtime:ro'
+    ]
+
+    if os.environ.get('JENKINS_GCE_SSH_PRIVATE_KEY_FILE'):
```

Comment:
When would we want to run without this? Please be more opinionated as it makes these simpler and easy to understand.